android/src/main/java/com/reactnativestripesdk/pushprovisioning/PushProvisioningProxy.kt

```diff
@@ -0,0 +1,109 @@
+package com.reactnativestripesdk
+
+import android.app.Activity
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import android.util.Log
+import com.facebook.react.bridge.*
+import com.reactnativestripesdk.pushprovisioning.AddToWalletButtonView
+import com.reactnativestripesdk.pushprovisioning.EphemeralKeyProvider
+import com.reactnativestripesdk.pushprovisioning.TapAndPayProxy
+import com.stripe.android.pushProvisioning.PushProvisioningActivity
+import com.stripe.android.pushProvisioning.PushProvisioningActivityStarter
+
+
+object PushProvisioningProxy {
+  private const val TAG = "StripePushProvisioning"
+  private var description = "Added by Stripe"
+  private var tokenRequiringTokenization: ReadableMap? = null
+
+  fun getApiVersion(): String {
+    return try {
+      Class.forName("com.stripe.android.pushProvisioning.PushProvisioningActivity")
+      PushProvisioningActivity.API_VERSION
+    } catch (e: Exception) {
+      Log.e(TAG, "PushProvisioning dependency not found")
+      ""
+    }
+  }
+
+  fun invoke(
+          context: ReactApplicationContext,
+          view: AddToWalletButtonView,
+          cardDescription: String,
+          ephemeralKey: String,
+          token: ReadableMap?
+  ) {
+    try {
+      Class.forName("com.stripe.android.pushProvisioning.PushProvisioningActivityStarter")
+      description = cardDescription
+      tokenRequiringTokenization = token
+      createActivityEventListener(context, view)
+      DefaultPushProvisioningProxy().beginPushProvisioning(
+        context.currentActivity!!,
+        description,
+        EphemeralKeyProvider(ephemeralKey)
+      )
+    } catch (e: Exception) {
+      Log.e(TAG, "PushProvisioning dependency not found")
+    }
+  }
+
+  fun isCardInWallet(activity: Activity, cardLastFour: String, promise: Promise) {
+    TapAndPayProxy.invoke(activity, cardLastFour, promise)
+  }
+
+  private fun createActivityEventListener(context: ReactApplicationContext, view: AddToWalletButtonView) {
+    val listener = object : BaseActivityEventListener() {
+      override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(activity, requestCode, resultCode, data)
+        if (requestCode == TapAndPayProxy.REQUEST_CODE_TOKENIZE) {
+          view.dispatchEvent(
+            if (resultCode == RESULT_OK) null else mapError("Failed", "Failed to resolve yellow path.", null, null, null, null)
```

Comment:
This is google's terminology for that particular state- https://developers.google.com/pay/issuers/apis/push-provisioning/android/wallet-operations#resolving_yellow_path. Maybe better to change it to something like "Failed to verify identity" since this path is only encountered when the token state is `TOKEN_STATE_NEEDS_IDENTITY_VERIFICATION`